src/components/Banner/Banner.tsx

```diff
@@ -2,108 +2,205 @@ import clsx from 'clsx';
    */
-  dismissible?: boolean;
+  dismissable?: boolean;
   /**
-   * Name of banner SVG icon. Available options are: banner, error, warning-2, success
+   * Optional callback to be triggered when the banner is dismissed
    */
-  iconName?: IconName;
+  onDismiss?: () => void;
   /**
-   * The title attribute for the SVG icon in the banner
+   * This is deprecated and will be removed in an upcoming release. Please use the default elevation 1
+   *
+   * The perceived elevation of the banner. An elevation of 0 appears flat against the surface while
+   * an elevation of 1 appears to hover slightly. The hover appearance is used to separate the element
+   * from the surrounding area. The flat version should only be used on white backgrounds.
+   *
+   * @deprecated
    */
-  iconTitle?: string;
+  isFlat?: boolean;
   /**
    * Controls the layout of the banner
    * - **vertical** renders the banner content center aligned and stacked
+   *
+   * Vertical banners are used in narrow areas, like sidebars
    */
   orientation?: 'vertical';
   /**
-   * Optional heading text for the banner that appears above banner body message
+   * The title/heading of the banner
+   */
+  title?: ReactNode;
+  /**
+   * The element the title renders as
    */
-  title?: string;
+  titleAs?: HeadingElement;
   /**
    * Stylistic variations for the banner type.
+   * - **brand** - results in a purple banner
+   * - **neutral** - results in a gray banner
    * - **success** - results in a green banner
    * - **warning** - results in a yellow banner
-   * - **error** - results in a red banner
-   * - **info** - results in a blue banner
+   * - **alert** - results in a red banner
    */
-  variant?: 'success' | 'warning' | 'error' | 'info' | 'brand';
+  variant?: 'brand' | 'neutral' | 'success' | 'warning' | 'alert';
 }
 
+const variantToIconAssetsMap: {
+  [key: string]: {
+    name: 'notifications' | 'forum' | 'check-circle' | 'warning' | 'dangerous';
+    title: string;
+  };
+} = {
+  brand: {
+    name: 'notifications',
+    title: 'attention',
+  },
+  neutral: {
+    name: 'forum',
+    title: 'notice',
+  },
+  success: {
+    name: 'check-circle',
+    title: 'success',
+  },
+  warning: {
+    name: 'warning',
+    title: 'warning',
+  },
+  alert: {
+    name: 'dangerous',
+    title: 'alert',
+  },
+};
+
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Banner} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A banner used to provide and highlight information to a user or ask for a decision or action.
+ *
+ * Example usage:
+ *
+ * ```tsx
+ * <Banner
+ *   onDismiss={handleDismiss}
+ *   text={
+ *     <>
+ *       <BannerTitle>Some title</Banner.Title>
+ *       <BannerDescription>Some description</Banner.Description>
+ *     </>
+ *   }
+ * />
+ * ```
  */
 export const Banner = ({
-  iconTitle,
+  action,
   className,
-  title,
-  children,
-  iconName,
-  dismissible,
+  description,
+  descriptionAs = 'p',
```

Comment:
nit: seems unused right now

Reply:
Whoops! Thanks for pointing that out! ❤️